packages/snap-controller/src/Autocomplete/AutocompleteController.test.ts

```diff
@@ -448,26 +466,28 @@ describe('Autocomplete Controller', () => {
 		});
 
 		await controller.bind();
+		let inputEl: HTMLInputElement | null;
 
-		const inputEl: HTMLInputElement | null = document.querySelector(controller.config.selector);
+		await waitFor(() => {
+			inputEl = document.querySelector(controller.config.selector);
+			expect(inputEl!).toBeDefined();
+		});
 
 		inputEl!.focus();
-		await new Promise((resolve) => setTimeout(resolve));
-
-		expect(document.activeElement).toBe(inputEl);
-
+		expect(document.activeElement).toBe(inputEl!);
 		const setFocusedfn = jest.spyOn(controller, 'setFocused');
 
 		inputEl!.dispatchEvent(new Event('click', { bubbles: true })); // should not loose focus
-
 		expect(setFocusedfn).not.toHaveBeenCalled();
-		expect(controller.store.state.focusedInput).toBe(inputEl);
-		expect(document.activeElement).toBe(inputEl);
+		await waitFor(() => {
+			expect(controller.store.state.focusedInput).toBe(inputEl!);
+		});
+		expect(document.activeElement).toBe(inputEl!);
 
 		const divEl = document.querySelector('div');
 		divEl!.dispatchEvent(new Event('click', { bubbles: true }));
 
-		expect(setFocusedfn).toHaveBeenCalledTimes(1);
+		expect(setFocusedfn).toHaveBeenCalledTimes(2);
```

Comment:
Nice to see this yielding the correct number of calls!

Reply:
Yeah!